File: biblioteca-web/src/app/models/book.model.ts
export interface Book {
  id: number;
  title: string;
  author: string;
  career: string;

  stock: number;
  libraryStock: number;
  totalCopies: number;
  availableCopies?: number;

  isbn: string;
  cover: string;

  isActive?: boolean;
}

File: backend/biblioteca-api/app/models/book_model.py
from pydantic import BaseModel, Field


class BookCreate(BaseModel):
    title: str = Field(..., min_length=1)
    author: str = Field(..., min_length=1)
    career: str = Field(..., min_length=1)
    isbn: str = Field(..., min_length=1)
    cover: str = Field(..., min_length=1)

    stock: int
    libraryStock: int
    totalCopies: int
    availableCopies: int

    isActive: bool = True